packages/nimble-components/src/table/specs/data-hierarchy-hld.md

```diff
@@ -59,6 +53,53 @@ interface TableRowExpansionToggleEventDetail {
 
 Note: This event will _not_ be emitted for group rows.
 
+### `Table` API for specifying lazy loaded hierarchical data
+
+In some cases, a client might know that a given row has children, but for performance reasons, that data hasn't been loaded from the backend and given to the table. To enable this use case, there will be an API on the table to specify that a given row should be expandable even if it has no children within the data and to specify that a row is loading its children. That API will look as follows:
+
+```ts
+interface TableRowOptions {
+    // Specifies whether or not the row should show the expand-collapse button
+    // even if it does not have any children in the data.
+    forceExpandable: boolean;
+
+    // Specifies whether or not the row should be in a state of loading its
+    // children.
+    loading: boolean;
+}
+
+public Table() {
+    ...
+    // Sets the options for the rows specified by the passed IDs.
+    public async setRowOptions(rowOptions: { id: string, options: TableRowOptions }[]): Promise<void>;
```

Comment:
I struggled to come up with a name for the type of `{ id: string, options: TableRowOptions }`, so I've just inlined the type in the function signature.

I'm open to names for this type to create an exported interface for it, but I also think we could proceed with the type only specified here.

Reply:
Closing as this is now outdated.